admin/resources/js/components/H2Components/Select.tsx

```diff
@@ -0,0 +1,123 @@
+import { QuestionMarkCircleIcon, XCircleIcon } from "@heroicons/react/solid";
+import React, { useState } from "react";
+import { RegisterOptions, useFormContext } from "react-hook-form";
+import InputContext from "./InputContext";
+import InputError from "./InputError";
+
+export interface SelectProps
+  extends React.SelectHTMLAttributes<HTMLSelectElement> {
+  /** HTML id used to identify the element. */
+  id: string;
+  /** The text for the label associated with the select input. */
+  label: string;
+  /** A string specifying a name for the input control. */
+  name: string;
+  /** List of options for the select element. */
+  options: { value: string | number; text: string }[];
+  /** Object set of validation rules to impose on input. */
+  rules?: RegisterOptions;
+  /** Optional context which user can view by toggling a button. */
+  context?: string;
+  /** If an error string is provided, it will appear below the fieldset inputs. */
+  error?: string;
```

Comment:
You don't need to pass error in, as we get errors from react-hook-form's useFormContext() .